x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java

```diff
@@ -79,6 +80,9 @@ public static List<Object[]> args() throws Exception {
 
     protected FieldExtractorTestCase(MappedFieldType.FieldExtractPreference preference) {
         this.preference = preference;
+        if (preference != null) {
+            assumeTrue("Requires pragma", Build.current().isSnapshot());
```

Comment:
Is there a capability that could b used instead, to check if it's enabled rather than checking for snapshot directly? That'd guard against forgetting to update the tests later.